src/components/Recipe.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { Link, useParams, useHistory } from "react-router-dom";
+import {
+  Title,
+  Button,
+  CircleList,
+  CircleListItem,
+  Container,
+} from "./Styles.js";
+import api from "../data/DataAPI.js";
+
+function IngredientList(props) {
+  const { ingredients } = props;
+  return (
+    <CircleList>
+      {ingredients &&
+        ingredients.map((ingredient) => (
+          <CircleListItem key={ingredient.name}>
+            {ingredient.name}
+          </CircleListItem>
+        ))}
+    </CircleList>
+  );
+}
+
+function RecipeDetail({ recipe }) {
+  const history = useHistory();
+
+  const deleteRecipe = () => {
+    let url = `${process.env.REACT_APP_API_BASEURL}/api/recipe/recipes/${recipe.id}/`;
+
+    fetch(url, { method: "DELETE" }).then(() => history.push("/recipes/"));
+  };
+
+  return (
+    <div>
+      {recipe ? (
+        <div>
+          <Title data-testid="recipe_title">{recipe.name}</Title>
+          <p data-testid="recipe_description">{recipe.description}</p>
+          <h3>Ingredients</h3>
+          <Container>
+            <IngredientList ingredients={recipe.ingredients} />
+          </Container>
+          <Container>
+            <Link to="/recipes">
+              <Button primary>Recipes</Button>
+            </Link>
+            <Link
+              to={{
+                pathname: `/recipes/${recipe.id}/edit`,
+              }}
+            >
+              <Button>Edit</Button>
+            </Link>
+            <Button onClick={deleteRecipe}>Delete</Button>
+          </Container>
+        </div>
+      ) : (
+        <Title>Recipe not found</Title>
+      )}
+    </div>
+  );
+}
+
+function Recipe(props) {
+  const { recipeId } = useParams();
+
+  const [{ data, error, isLoading }, dispatch] = useState({
```

Comment:
why are you using dispatch here?

Reply:
Are you referring to the method naming? setState or similar will be clear, right?

Reply:
exactly, not a big deal but this naming can create confusion with the **useReducer** hook